extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -10,46 +10,79 @@ class NodeImage extends Node {
 	const CAPTION_TAG_NAME = 'caption';
 	const MEDIA_TYPE_VIDEO = 'VIDEO';
 
+	private function getImage($key, $caption, $alt) {
+		$title = $this->getImageAsTitleObject( $key );
+		$file = $this->getFilefromTitle( $title );
+		if ( $title instanceof \Title ) {
+			$this->getExternalParser()->addImage( $title->getDBkey() );
+		}
+		$ref = null;
+		wfRunHooks( 'PortableInfoboxNodeImage::getData', [ $title, &$ref, $caption ] );
+		$item = [
+			'url' => $this->resolveImageUrl( $file ),
+			'name' => ( $title ) ? $title->getText() : '',
+			'key' => ( $title ) ? $title->getDBKey() : '',
+			'alt' => $alt,
+			'caption' => $caption,
+			'ref' => $ref
+		];
+		if ( $this->isVideo( $file ) ) {
+			$item = $this->videoDataDecorator( $item, $file );
+		}
+		return $item;
+	}
+
 	public function getData() {
 		if ( !isset( $this->data ) ) {
 			$imageData = $this->getRawValueWithDefault( $this->xmlNode );
-
-			if( is_string($imageData) && PortableInfoboxDataBag::getInstance()->getGallery($imageData)) {
-				$imageData = PortableInfoboxDataBag::getInstance()->getGallery($imageData);
-			}
-
-			$title = $this->getImageAsTitleObject( $imageData );
-			$file = $this->getFilefromTitle( $title );
-			if ( $title instanceof \Title ) {
-				$this->getExternalParser()->addImage( $title->getDBkey() );
-			}
-			$ref = null;
-			$alt = $this->getValueWithDefault( $this->xmlNode->{self::ALT_TAG_NAME} );
-			$caption = $this->getValueWithDefault( $this->xmlNode->{self::CAPTION_TAG_NAME} );
-
-			wfRunHooks( 'PortableInfoboxNodeImage::getData', [ $title, &$ref, $caption ] );
-
-			$this->data = [
-				'url' => $this->resolveImageUrl( $file ),
-				'name' => ( $title ) ? $title->getText() : '',
-				'key' => ( $title ) ? $title->getDBKey() : '',
-				'alt' => $alt,
-				'caption' => $caption,
-				'ref' => $ref
-			];
-
-			if ( $this->isVideo( $file ) ) {
-				$this->data = $this->videoDataDecorator( $this->data, $file );
+			$imageDataLower = strtolower($imageData);
+
+			if ( strpos( $imageDataLower, '-tabber-' ) !== false || strpos( $imageDataLower, '-gallery-' ) !== false ) {
```

Comment:
Thank you for your feedback.

I think code could be way cleaner if we had parsers for galleries and tabbers that parse them to some sort of intermidate data structures instead of HTML right away. Unfortunetly those parsers currently do not exists and creating them would have extra, possibly high, cost - I think it's important to point out here that such parser would be way easier to create for gallery [which is just list of images with captions] but in case of tabber situation is more complicated - cause tabber can have any wikitext inside it.

Given the situation: nor galleries nor tabber is under active development, and feature we are developing is "experimental" I suggest we stick to taken approach and review it in the future depending on the future shape of this feature. In order to become more "immune" to gallery parsing changes we should implement solid unit tests.

Thoughts?

Btw. approach with PortableInfoboxDataBag is interesting, but note that it doesn't work in case when ```<gallery>``` (or ```<tabber>```) is wrapped in ```<center>``` for instance, e.g.: ```image=<center><tabber>...</tabber></center>``` because then in Node you get ```<center>UNIQUE...</center>``` as a parameter value (instead of just the ```UNIQUE...```) so you can't retrieve it from the data bag (at least not without extra processing).